geo-types/src/geometry.rs

```diff
@@ -234,3 +237,206 @@ where
         Geometry::Triangle(_) => type_name::<Triangle<T>>(),
     }
 }
+
+#[cfg(any(feature = "approx", test))]
+impl<T> RelativeEq for Geometry<T>
+where
+    T: AbsDiffEq<Epsilon = T> + CoordNum + RelativeEq,
+{
+    #[inline]
+    fn default_max_relative() -> Self::Epsilon {
+        T::default_max_relative()
+    }
+
+    /// Equality assertion within a relative limit.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use geo_types::{Geometry, polygon};
+    ///
+    /// let a: Geometry<f32> = polygon![(x: 0., y: 0.), (x: 5., y: 0.), (x: 7., y: 9.), (x: 0., y: 0.)].into();
+    /// let b: Geometry<f32> = polygon![(x: 0., y: 0.), (x: 5., y: 0.), (x: 7.01, y: 9.), (x: 0., y: 0.)].into();
+    ///
+    /// approx::assert_relative_eq!(a, b, max_relative=0.1);
+    /// approx::assert_relative_ne!(a, b, max_relative=0.001);
+    /// ```
+    ///
+    fn relative_eq(
+        &self,
+        other: &Self,
+        epsilon: Self::Epsilon,
+        max_relative: Self::Epsilon,
+    ) -> bool {
+        match self {
+            Geometry::Point(g) => {
+                if let Geometry::Point(other) = other {
+                    g.relative_eq(other, epsilon, max_relative)
+                } else {
+                    false
+                }
+            }
```

Comment:
Another way to format this, if you prefer it:

```suggestion
        match (self, other) {
            (Geometry::Point(g1), Geometry::Point(g2)) => g1.relative_eq(g2, epsilon, max_relative),
            ...
            _ => false,
        }
```

Reply:
This is better! I'll apply this to the rest of the cases.

Reply:
done!